### PR TITLE
ci: add workflow_dispatch so build can be triggered from the move2kube repo

### PR DESCRIPTION
**Pull request by HarikrishnanBalagopal**
Signed-off-by: Harikrishnan Balagopal <harikrishmenon@gmail.com>

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 name: Build
 
 on:
+  workflow_dispatch: {}
```
Comment (ashokponkumar) [inline .github/workflows/build.yml:4]: Doesn't it need the branch info on which branch it is getting triggered?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
...
   push:
     branches:
       - "main"
```
Comment (ashokponkumar) [inline .github/workflows/build.yml:7]: We should probably remove this "main" filter too.

**Review by ashokponkumar — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (HarikrishnanBalagopal) [reply .github/workflows/build.yml:4]: yes it gets that from `context.ref`

File: .github/workflows/build.yml (hunk 2 above)
Comment (HarikrishnanBalagopal) [reply .github/workflows/build.yml:7]: why?

**Review by ashokponkumar — APPROVED** [review]

**Pull request merged**